relay-server/src/actors/project_upstream.rs

```diff
@@ -261,7 +266,10 @@ impl UpstreamProjectSource {
                                         Some(ProjectState::err())
                                     })
                                     .unwrap_or_else(ProjectState::missing);
-
+                                metric!(
+                                    histogram(RelayHistograms::ProjectStateAttempts) =
+                                        channel.attempts
+                                );
```

Comment:
How does this work with retries? If relay makes an attempt and fails, this will record an attempt. If relay fails the following attempt, does this record `2`? So `1 + 2 = 3`, when it should be `1 + 1 = 2`.

Reply:
Because of the `continue` on line 253, this code only executes when the endpoint returned an actual state in the `configs` object. The actual arithmetic on `attempts` happens on line 203: https://github.com/getsentry/relay/blob/d565bca404a41a3e170e95c40faa02f909dfdc93/relay-server/src/actors/project_upstream.rs#L203

When a project key is in the `pending` list, it is re-inserted into `state_channels`, so it will increment the count: https://github.com/getsentry/relay/blob/d565bca404a41a3e170e95c40faa02f909dfdc93/relay-server/src/actors/project_upstream.rs#L252